github-runner-manager/src/github_runner_manager/platform/jobmanager_provider.py

```diff
@@ -168,56 +174,43 @@ def get_runner_context(
             PlatformApiError: Problem with the underlying API.
 
         Returns:
-            New runner token.
+            A tuple containing the runner context and the self-hosted runner.
         """
-        configuration = jobmanager_client.Configuration(host=self._url)
-        with jobmanager_client.ApiClient(configuration) as api_client:
-            api_instance = jobmanager_client.RunnersApi(api_client)
-            try:
-                runner_register_request = jobmanager_client.RunnerCreate(
-                    name=instance_id.name, labels=labels
-                )
-
-                response = api_instance.register_runner_v1_runners_register_post(
-                    runner_register_request
-                )
-                updated_metadata = RunnerMetadata(
-                    platform_name=metadata.platform_name, url=self._url
+        try:
+            response = self._jobmanager_api.register_runner(name=instance_id.name, labels=labels)
+            updated_metadata = RunnerMetadata(platform_name=metadata.platform_name, url=self._url)
+            updated_metadata.runner_id = str(response.id)
+            if token := response.token:
+                jobmanager_endpoint = f"{self._url}/v1/runners/{updated_metadata.runner_id}/health"
+                # For now, use the first label
+                label = "undefined"
+                if labels:
+                    label = labels[0]
+                command_to_run = (
+                    f"BUILDER_LABEL={label} JOB_MANAGER_BEARER_TOKEN={token} "
+                    f"JOB_MANAGER_API_ENDPOINT={jobmanager_endpoint} "
+                    "builder-agent"
                 )
-                updated_metadata.runner_id = str(response.id)
-                if token := response.token:
-                    jobmanager_endpoint = (
-                        f"{self._url}/v1/runners/{updated_metadata.runner_id}/health"
-                    )
-                    # For now, use the first label
-                    label = "undefined"
-                    if labels:
-                        label = labels[0]
-                    command_to_run = (
-                        f"BUILDER_LABEL={label} JOB_MANAGER_BEARER_TOKEN={token} "
-                        f"JOB_MANAGER_API_ENDPOINT={jobmanager_endpoint} "
-                        "builder-agent"
-                    )
-                    return (
-                        RunnerContext(
-                            shell_run_script=command_to_run,
-                            ingress_tcp_ports=[8080],
+                return (
+                    RunnerContext(
+                        shell_run_script=command_to_run,
+                        ingress_tcp_ports=[8080],
```

Comment:
It looks like we're hardcoding 8080, should this be in the setup of cloud security group during the initialization of the application?

Reply:
I think the way this was done was historically necessary to dynamically support JobManager vs. GitHub mode depending on the messages in the queue. We could do it during application initialization now since the mode is predetermined. However, I think it's out of scope for this PR.

Reply:
Fine with me, I'll add this to the refactor notes.